src/App.jsx

```diff
@@ -14,45 +16,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(goods, sortField, isReversed) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState('');
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, isReversed);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
+          className={cn(
+            { 'is-light': sortField !== SORT_FIELD_ALPHABET },
+            'button is-info',
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          className={cn(
+            { 'is-light': sortField !== SORT_FIELD_LENGTH },
+            'button is-success',
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setIsReversed(prevState => !prevState)}
+          className={cn(
+            { 'is-light': !isReversed },
+            'button is-warning',
+          )}
+        >
+          Reverse
+        </button>
+
+        {(sortField || isReversed) && (
+        <button
+          type="button"
+          onClick={() => {
+            setSortField('');
+            setIsReversed(false);
+          }}
+          className="button is-danger"
```

Comment:
The button should be lighter. No need to delete the `is-light` class.

![image](https://github.com/mate-academy/react_list-of-goods-js/assets/116377279/ab4768c6-881a-4cd1-975e-8b7a12518e57)

Reply:
➕ , reset button should look like that
![image](https://github.com/mate-academy/react_list-of-goods-js/assets/96409563/16a958fe-cb20-47ce-aeea-895a644dde42)